src/main/java/io/jenkins/plugins/oidc_provider/IdTokenCredentials.java

```diff
@@ -133,18 +150,27 @@ public final String getAudience() {
         this.audience = Util.fixEmpty(audience);
     }
 
-    protected abstract IdTokenCredentials clone(KeyPair kp, Secret privateKey);
+    @DataBoundSetter public void setAlgorithm(SupportedKeyAlgorithm algorithm) {
+        Objects.requireNonNull(algorithm);
+        this.algorithm = algorithm;
+    }
+
+    @NonNull public SupportedKeyAlgorithm getAlgorithm() {
+        return algorithm;
+    }
```

Comment:
Propose doing #18 first, and then moving this option to `IdTokenConfiguration`, as I doubt there is a legitimate use case for a single controller having tokens with two issuers that use different algorithms; and a Jenkins admin would probably prefer to force use of a sufficiently secure algorithm.

Reply:
The only reason I could think of to not make the algorithm global would be if certain
providers only support a subset of algorithms. Then it might make sense to keep it.

But then, I am by no means a expert in OIDC so maybe this is an unfounded fear :smile:

Reply:
I think everyone supports RS256 and IIRC it is mandated. Beyond that I am unsure.

Reply:
What exactly do you mean by that? :thinking: 
That all keys *must* be signed by a RS256 key?

Reply:
No, that all OIDC impls must support this algorithm at a minimum.

Reply:
Just to clarify, you are suggesting making the algorithm used for the keypair a global configuration, am I right?

Reply:
Right; #18 provides the global config class for another purpose, so if merged, that would be a place to move this option.

Reply:
As said, I am not 100% sure if it makes sense to force the whole Jenkins instance to use a certain algorithm
for generating the key pair.

Reply:
> [It is possible] that certain OIDC connectors doesn't support elliptic curve key pairs

Right but in such a case why not stick with RS256, which all requesting parties should support? One possible reason is because you are a Jenkins admin who does not consider RS256 sufficiently secure and you want to force use of stronger cryptography. That would be undermined if the algorithm were overridable by developers in charge of some folders.

If there is no admin-vs.-dev distinction—one person or group has full config access to Jenkins—then it would make sense to switch to (say) ES512 to connect to one service that supports it, but stick with RS256 for another service which does not.

Not sure at this point what is best, just trying to think through the implications of letting the algorithm be configurable by people who are not Jenkins administrators. Of course all of the new options are stronger than the currently hard-coded option.

Reply:
You are right, I guess that behavior will anyway be an edge case :slightly_smiling_face: 
I'll wait for your PR to be merged then and integrate my changes in the global configuration :slightly_smiling_face:

Reply:
> wait for your PR to be merged

Merged (and released).